### PR TITLE
React movies list - fetch movies

### PR DESCRIPTION
**Pull request by vsupruniuk**
[DEMO LINK](https://vsupruniuk.github.io/react_movies-list-fetch-movies/)

### CODE REVIEW

File: src/components/FindMovie/FindMovie.tsx [hunk 1]
```diff
@@ -1,54 +1,111 @@
...
-        </div>
+type Props = {
+  addMovie: (movie: Movie) => void,
+};
 
-        <p className="help is-danger">
-          Can&apos;t find a movie with such a title
-        </p>
-      </div>
+export const FindMovie: React.FC<Props> = ({ addMovie }) => {
```
Comment (jstmpelowycz) [inline src/components/FindMovie/FindMovie.tsx:12]: apply `memo`

File: src/components/FindMovie/FindMovie.tsx [hunk 2]
```diff
@@ -1,54 +1,111 @@
...
+  const [query, setQuery] = useState('');
+  const [error, setError] = useState('');
+  const [movie, setMovie] = useState<Movie | null>(null);
+
+  const onTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
```
Comment (jstmpelowycz) [inline src/components/FindMovie/FindMovie.tsx:17]: when creating such callback, consider applying `useCallback`

File: src/App.tsx [hunk 3]
```diff
@@ -1,29 +1,27 @@
...
-interface State {
-  movies: Movie[];
-}
+import './App.scss';
 
-export class App extends Component<{}, State> {
-  state: State = {
-    movies: [],
-  };
+export const App: React.FC = () => {
```
Comment (jstmpelowycz) [inline src/App.tsx:8]: `memo`?

**Review by jstmpelowycz — CHANGES_REQUESTED** [review]

File: src/api/api.ts [hunk 4]
```diff
@@ -0,0 +1,8 @@
+import { Movie } from '../react-app-env';
+
+const API_URL = ' https://www.omdbapi.com/?apikey=830ad734&t=';
+
+export function getMovie(title: string) :Promise<Movie> {
```
Comment (illiarerun) [inline src/api/api.ts:5]:
```suggestion
export function getMovie(title: string): Promise<Movie> {
```

File: src/api/api.ts [hunk 5]
```diff
@@ -0,0 +1,8 @@
+import { Movie } from '../react-app-env';
+
+const API_URL = ' https://www.omdbapi.com/?apikey=830ad734&t=';
```
Comment (illiarerun) [inline src/api/api.ts:3]:
```suggestion
const API_URL = 'https://www.omdbapi.com/?apikey=830ad734&t=';
```

File: src/components/FindMovie/FindMovie.tsx [hunk 6]
```diff
@@ -1,54 +1,113 @@
...
+    getMovie(query)
+      .then((response: Movie) => {
+        if (response.Error) {
+          setMovie(null);
+          setError(response.Error);
+        } else {
+          setMovie(response);
+          setError('');
+        }
+      });
```
Comment (illiarerun) [inline src/components/FindMovie/FindMovie.tsx:36]: it'd be easier and more clear to use `async/await` syntax

File: src/components/FindMovie/FindMovie.tsx [hunk 7]
```diff
@@ -1,54 +1,113 @@
...
+  }, []);
+
+  const handleMovieAdd = () => {
+    if (movie) {
+      addMovie(movie);
+    }
```
Comment (illiarerun) [inline src/components/FindMovie/FindMovie.tsx:42]: Clear movie after adding it
image

**Review by illiarerun — CHANGES_REQUESTED** [review]

**Review by illiarerun — APPROVED** [review]

**Pull request closed without merging**